### PR TITLE
Gps functionality

### PR DESCRIPTION
**Pull request by jamiesanson**
This PR contains code to get the GPS working. Current locking information is lacking, as the library in use doesn't handle GSV NMEA messages meaningfully, but is information we could scan for ourselves in future

### CODE REVIEW

File: src/teensy_sketch/teensy_sketch.ino [hunk 1]
```diff
@@ -74,6 +74,7 @@
 #define dataFileName "data/data"
 #define initExtension "txt"
 #define dataExtension "csv"
+#define csvLayout "ACCEL_X,ACCEL_Y,ACCEL_Z,GYRO_X,GYRO_Y,GYRO_Z,MAG_X,MAG_Y,MAG_Z,LAT,LNG,"
```
Comment (marcelvanworkum) [inline src/teensy_sketch/teensy_sketch.ino:77]: is the trailing `,` needed?

File: src/teensy_sketch/teensy_sketch.ino [hunk 2]
```diff
@@ -98,41 +99,63 @@ boolean L3G4200DReadReady = false;
...
+  readGPS();
   readIMU();
-  readGyro();
-  
-  delay(200);
-  dataModule.println(getIMULogString(accelData)+getIMULogString(gyroData)+getIMULogString(magData));
-  dataModule.println(getGyroLogString(L3G4200DData));
+
+  // Print all information
+  dataModule.print(getIMULogString(accelData)+getIMULogString(gyroData)+getIMULogString(magData));
```
Comment (marcelvanworkum) [inline src/teensy_sketch/teensy_sketch.ino]: spaces around `+` to keep consistent with rest of the file

File: src/teensy_sketch/teensy_sketch.ino [hunk 3]
```diff
@@ -176,3 +243,9 @@ String getIMULogString(float * data) {
 String getGyroLogString(int16_t * data) {
   return (String(data[0], DEC) + "," + String(data[1], DEC) + "," + String(data[2], DEC) + ",");
 }
+
+// Function for returning formatted GPS string
+String getGPSLogString() {
+  return (String(gps.location.lat(),6) + "," + String(gps.location.lng(),6) + ",");
```
Comment (marcelvanworkum) [inline src/teensy_sketch/teensy_sketch.ino]: space between second argument of #String `, 6`.

Also 6 is a magic number, I assume it's to limit the places of precision right?

File: src/teensy_sketch/teensy_sketch.ino (hunk 1 above)
Comment (jamiesanson) [reply src/teensy_sketch/teensy_sketch.ino:77]: Yep. Indicates next cell for `.csv` format

File: src/teensy_sketch/teensy_sketch.ino (hunk 2 above)
Comment (jamiesanson) [reply src/teensy_sketch/teensy_sketch.ino]: On it

File: src/teensy_sketch/teensy_sketch.ino (hunk 3 above)
Comment (jamiesanson) [reply src/teensy_sketch/teensy_sketch.ino]: Yep, decimal place precision. Will make it a constant though as we may use different number rounding elsewhere

**Comment by marcelvanworkum** [comment]
lgtm 🎱

**Pull request merged**